bitcoin/src/bip32.rs

```diff
@@ -44,7 +45,7 @@ impl_array_newtype_stringify!(ChainCode, 32);
 
 impl ChainCode {
     fn from_hmac(hmac: Hmac<sha512::Hash>) -> Self {
-        hmac.as_ref()[32..].try_into().expect("half of hmac is guaranteed to be 32 bytes")
+        ChainCode(*hmac.as_byte_array().sub_array::<32, 32>())
```

Comment:
In b04b2b8c58cc7770ed7ecf8507e4e380563976d8:

I would prefer to use `.split_array::<32, 32>().1` to emphasize that the hmac has exactly 64 bytes and you're taking exactly the second half of it. (Similar everywhere you are using `sub_array::<0, 32>` or `sub_array::<32, 32>`.)

Reply:
Oh, yes, good idea.